catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
+    }
+
+    public String getRootBackupDir() {
+        return rootBackupDir.getAbsolutePath();
+    }
+
+    /**
+     * Sets the root file system backup directory.
+     *
+     * @param dir absolute path for the root file system backup directory.
+     */
+    public void setRootBackupDir(String dir) {
+        if (StringUtils.isBlank(dir)) {
+            LOGGER.error("The root backup directory is blank.");
+            return;
+        }
+
+        this.rootBackupDir = new File(dir);
+        LOGGER.debug("Set root backup directory to: {}", this.rootBackupDir.toString());
+    }
+
+    public PeriodicBatchExecutor<Metacard> getPeriodicDrain() {
+        return periodicDrain;
+    }
+
+    public void setPeriodicDrain(PeriodicBatchExecutor<Metacard> periodicDrain) {
+        this.periodicDrain = periodicDrain;
+        getPeriodicDrain().execute(this::backup);
+    }
+
```

Comment:
I know you didn't touch this code but would it be possible to do one of two things: 1) get rid of the enum and use simple strings, that way we can get rid of the switch statements; 2) rename the enum to `Operation` so it at least follows our naming conventions.